tree/dataframe/src/RSqliteDS.cxx

```diff
@@ -26,6 +32,328 @@
+   int flags,
+   int *pResOut)
+   {
+      *pResOut = 0;
+      if (flags == SQLITE_ACCESS_READWRITE) {
+         return SQLITE_OPEN_READONLY;
+      }
+   return SQLITE_OK;
+   }
+
+int VfsRdOnlyFullPathname(
+   sqlite3_vfs * /*vfs*/,
+   const char *zPath,
+   int nOut,
+   char *zOut)
+{
+   zOut[nOut-1] = '\0';
+   sqlite3_snprintf(nOut, zOut, "%s", zPath);
+   return SQLITE_OK;
+}
+
+/**
+ * Taken from unixRandomness
+ */
+static int VfsRdOnlyRandomness(
+   sqlite3_vfs * /*vfs*/,
+   int nBuf,
+   char *zBuf)
+{
+   assert((size_t)nBuf >= (sizeof(time_t) + sizeof(int)));
+   memset(zBuf, 0, nBuf);
+   pid_t randomnessPid = getpid();
+   int fd, got;
+   fd = open("/dev/urandom", O_RDONLY, 0);
+   if (fd < 0) {
+     time_t t;
+     time(&t);
+     memcpy(zBuf, &t, sizeof(t));
+     memcpy(&zBuf[sizeof(t)], &randomnessPid, sizeof(randomnessPid));
+     assert(sizeof(t) + sizeof(randomnessPid) <= (size_t)nBuf);
+     nBuf = sizeof(t) + sizeof(randomnessPid);
+   } else {
+     do {
+       got = read(fd, zBuf, nBuf);
+     } while (got < 0 && errno == EINTR);
+     close(fd);
+   }
+  return nBuf;
+}
+
+static int VfsRdOnlySleep(
+   sqlite3_vfs * /*vfs*/,
+   int microseconds)
+{
+   gSystem->Sleep(microseconds / 1000);
+   return microseconds;
+}
+
+static int VfsRdOnlyGetLastError(
+   sqlite3_vfs * /*vfs*/,
+   int /*not_used1*/,
+   char * /*not_used2*/)
+{
+   return 0;
+}
+
+/**
+ * Taken from unixCurrentTimeInt64()
+ */
+static int VfsRdOnlyCurrentTimeInt64(
+   sqlite3_vfs * /*vfs*/,
+   sqlite3_int64 *piNow)
+{
+   static const sqlite3_int64 unixEpoch = 24405875*(sqlite3_int64)8640000;
+   int rc = SQLITE_OK;
+   struct timeval sNow;
+   if (gettimeofday(&sNow, 0) == 0) {
+      *piNow = unixEpoch + 1000*(sqlite3_int64)sNow.tv_sec + sNow.tv_usec/1000;
+   } else {
+      rc = SQLITE_ERROR;
+   }
+   return rc;
+}
+
+/**
+ * Taken from unixCurrentTime
+ */
+static int VfsRdOnlyCurrentTime(
+   sqlite3_vfs *vfs,
+   double *prNow)
+{
+   sqlite3_int64 i = 0;
+   int rc = VfsRdOnlyCurrentTimeInt64(vfs, &i);
+   *prNow = i/86400000.0;
+   return rc;
+}
+
+
+static struct sqlite3_vfs kSqlite3_vfs = {
+   3,
+   sizeof(VfsRootFile),
+   PATH_MAX,
+   NULL,  /* TODO sqlite3_vfs *pNext ?? */
+   gSQliteVfsName,
+   NULL, /* app data */
+   VfsRdOnlyOpen,
+   VfsRdOnlyDelete,
+   VfsRdOnlyAccess,
+   VfsRdOnlyFullPathname, // to do
+   NULL,
+   NULL,
+   NULL,
+   NULL,
+   VfsRdOnlyRandomness, // eventually --> TODO TRandom calls
+   VfsRdOnlySleep,
+   VfsRdOnlyCurrentTime,
+   VfsRdOnlyGetLastError,
+   VfsRdOnlyCurrentTimeInt64,
+   NULL,
+   NULL,
+   NULL
+};
+
+static bool Register() {
+   int retval;
+   retval = sqlite3_vfs_register(&kSqlite3_vfs, false);
+   return (retval == SQLITE_OK);
+}
+
+static bool IsURL(std::string_view fileName) {
+   auto haystack = std::string(fileName);
+   if (haystack.find ("http://")==0)
+      return true;
+   if (haystack.find ("https://")==0)
+      return true;
+   return false;
+}
```

Comment:
Please consider using `compare()` here instead of `find()`.
https://en.cppreference.com/w/cpp/string/basic_string/compare